### PR TITLE
Apply refact plugin

### PR DESCRIPTION
**Pull request by alanz**

### CODE REVIEW

**Comment by alanz** [comment]
Closes #131

File: stack.yaml [hunk 1]
```diff
@@ -13,6 +14,10 @@ packages:
     commit: b9bd4ebf77b22d2d9061d647d7799ddcc7c51228
     # commit: bff86be69f556f80a8dcd9dd42774ab77cb00eba
   extra-dep: true
+# - location:
+#     git: https://github.com/alanz/hlint.git
+#     commit: 6fb531d5ba32a4ba0e1a1c189e5f81ac7c43803f
+#   extra-dep: true
```
Comment (JPMoresmau) [inline stack.yaml]: Are these lines still needed for reference?

**Comment by JPMoresmau** [comment]
I see the value of applying hlint suggestions via refact, but what's the pipeline there? The ghc-mod lint command we expose return a Text, since ghc-mod applies show to the list of hlint suggestions. Since we seem to call hlint ourselves in that plugin, shouldn't we

- remove the ghc-mod lint command
- provide a lint command that returns a properly typed result, with proper location information?

File: stack.yaml (hunk 1 above)
Comment (alanz) [inline stack.yaml]: Nope, will delete them

**Comment by alanz** [comment]
Its a bit of a strange pipeline, but models what happens outside of hie.

In the normal case, ghc-mod or flyecheck or whatever runs hlint and displays the warnings.

Then [hlint-refactor-mode](https://github.com/mpickering/hlint-refactor-mode) which actually knows nothing about the displayed hints, invokes `hlint` itself, in an integrated pipeline mode which calls the `refactor` command from `apply-refact`.

So the two could potentially be collapsed, but have no specific requirement that this is so. Essentially the one runs hlint to display the changes, the other runs it in a different mode to apply one or more of them

File: hie-apply-refact/hie-apply-refact.cabal [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+category:            Web
+build-type:          Simple
+-- extra-source-files:
+cabal-version:       >=1.10
+
+library
+  exposed-modules:     Haskell.Ide.ApplyRefactPlugin
+  build-depends:       base >= 4.7 && < 5
+                     , aeson
+                     , apply-refact
```
Comment (JPMoresmau) [inline hie-apply-refact/hie-apply-refact.cabal]: I thought apply-refact was integrated into hlint, and we used it via hlint invocation, do we need that dependency? I don't see any usage of apply-refact modules.

Comment (alanz) [inline hie-apply-refact/hie-apply-refact.cabal]: See the end of the log in this build for what happens without this
dependency: https://travis-ci.org/alanz/haskell-ide-engine/builds/96739271

hlint does manage the application of the hints, but does so by calling the
`refactor` executable from `apply-refact`.

Comment (JPMoresmau) [inline hie-apply-refact/hie-apply-refact.cabal]: OK!

**Pull request merged**